File: console/cpp/src/Card.cpp
#include "Card.h"

using namespace std;
using namespace binary;
using namespace binaryAnte;


Card::Card(){
    id = time(0);
    value = Binary();
}
Card::Card(BinaryType type){
    id = time(0);
    value = Binary(type);
}
Card::Card(BinaryType type, Identifier identifier){
    id = identifier.nextId();
    value = Binary(type);
}

Card::~Card(){
}


long Card::getId(){
    return id;
}


string Card::toString(){
    return to_string(id) + ":" + value.toString();
}

File: console/CSharp/binary-ante/BinaryAnte/Card.cs
using System;
using VT.Types.Binary;

namespace VT.BinaryAnte
{
	public class Card
	{
		int Id;
		Binary Value;


		public Card(Identifier identifier){
			Id = identifier.NextId(Identifier.SIZE.HUGE2);
			Value = new Binary(Id);
		}
		public Card(BinaryType type, Identifier identifier){
			Id = identifier.NextId(Identifier.SIZE.LARGE);
			Value = new Binary(Id, type);
		}


		public int GetId(){
			return Id;
		}
		public Binary GetValue(){
			return Value.Clone();
		}


		public string ToString(){
			return Id + ":" + Value.ToString();
		}
	}
}
